tests/testproject/TestProject.AppHost/TestProgram.cs

```diff
@@ -66,10 +74,12 @@ private TestProgram(string[] args, Assembly assembly, bool includeIntegrationSer
                 .WithReference(kafka)
                 .WithReference(cosmos);
         }
+
+        AppBuilder.Services.TryAddLifecycleHook<EndPointWriterHook>();
```

Comment:
Why are we using the `Try*` variant here? The tests wouldn't work at all if this didn't get added, correct?

Reply:
The documentation didn't mention this bit - https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.lifecycle.lifecyclehookservicecollectionextensions.tryaddlifecyclehook?view=dotnet-aspire-8.0 . But I'm guessing that it's a well known pattern followed here.

Reply:
> But I'm guessing that it's a well known pattern followed here.

Correct. It's how DI works.

https://stackoverflow.com/questions/41743150/aspnet-core-di-tryadd-vs-add-usage

https://adrientorris.github.io/dotnet/dependency-injection-dotnet-difference-between-add-tryadd-tryaddenumerable-methods

Reply:
thanks!